scripts/buildLocales.js

```diff
@@ -0,0 +1,31 @@
+import * as fs from 'fs'
+
+// This script loads the current JSON-based locales files and merges them into
+// a single importable ES module for bundling purposes
+const localesRoot = process.argv[2]
+if (!localesRoot) {
+    console.log('usage: buildLocales <locales/feature dir>')
+}
+
+const locales = {}
+const localesDirs = fs.readdirSync(localesRoot).filter(f => !f.startsWith('.'))
+for (const l of localesDirs) {
+    locales[l] = {}
+    const dir = `${localesRoot}/${l}`
+    const files = fs.readdirSync(dir)
+    for (const f of files) {
+        const localeJSON = fs.readFileSync(`${dir}/${f}`)
+        const stringObj = JSON.parse(localeJSON)
+        locales[l][f] = {}
+        for (const [key, value] of Object.entries(stringObj)) {
+            if (key !== 'smartling') {
+                locales[l][f][key] = value.title
+            }
+        }
+    }
+}
+
+const output = 'export default `' +
+    JSON.stringify(locales).replace('`', '\\`') +
```

Comment:
I'm not sure the correct way to do this, but this line makes me nervous. I'm thinking the translations are untrusted input (came from a third-party service) and this output is going into our extension, so we should be careful to get escaping right here. What do you think @jonathanKingston?

Reply:
The intent here is not to prevent code injection, but to support the legitimate use of backticks in strings.

My understanding is that using element properties is safe (vs directly in content).  If we want to defense it depth we could go back to using the template literals but that was overkill on my part I think.

Reply:
Fair enough, but would you mind taking a quick look though @jonathanKingston? Happy to go with this if you think it's safe.

Reply:
I think this is probably 'good enough' for the reasons Lucas gives; this comes from a known source and we're trying to prevent legitimate use cases and data from breaking the script.

Escaping is hard and I'm sure there's something we're missing here but there's not a 'threat actor'.

If we wish to defend against this, writing a super simple test that imports the file and reads some properties from the translation would ensure the escaping isn't breaking once released.

Reply:
OK cool, if you're happy then it sounds good to me :+1: